### ISSUE TITLE
Atom stuck on 1.34.0-nightly9

### ISSUE THREAD

**Issue by wolfy1339** [atom/atom#18417]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

When I try to update Atom to 1.34.0-nightly12 it won't update and stays on 1.34.0-nightly9

### Steps to Reproduce

1. Go to the about page
2. Click on check for updates
**OR**
1. Download a fresh installer for 1.34.0-nightly12
2. Run it
3. Go to the about page, version is stuck at 1.34.0-nightly9

**Expected behavior:** For the upgrade to happen a
**Actual behavior:** 100%

**Reproduces how often:** [What percentage of the time does it reproduce?]

### Versions
```
$ atom --version

Atom    : 1.34.0-nightly9
Electron: 2.0.12
Chrome  : 61.0.3163.100
Node    : 8.9.3
```
```
$ apm --version
apm  2.1.3
npm  6.2.0
node 8.9.3 x64
atom 1.34.0-nightly9
python 2.7.15
git 2.19.1.windows.1
visual studio 2015
```

**Comment by rsese** [comment]
Thanks for the report - I believe this is a known issue with the nightlies on Windows though i couldn't find an issue for it so I'll leave it open for now /cc @daviwil

**Comment by rsese** [comment]
No progress, @daviwil is still OoO.  Nightly releases aren't yet officially announced and the work to be done before announcing is being tracked  in https://github.com/atom/atom/issues/17884.  I'll reference this issue there.

**Comment by daviwil** [comment]
TL;DR: the problem is in how [Squirrel.Windows](https://github.com/Squirrel/Squirrel.Windows) sorts prerelease version numbers, making `-nightly9` seem like a newer version number than `-nightly1*`, hence Atom Nightly on Windows never updates correctly past `-nightly9`.

Here's the details:

Squirrel.Windows is built upon [NuGet](https://www.nuget.org/) 2.0 which uses the [SemVer 1.0](https://semver.org/spec/v1.0.0.html) versioning scheme for packages, especially when it comes to prerelease versions.  This is why Atom Beta and Nightly version numbers look like `1.37.0-beta0`.

When NuGet originally implemented their SemVer 1.0 support, they didn't sort prerelease version numbers past 9 correctly like `1.37.0-nightly10`, etc.  Instead of doing a numerical comparison of the digits of a prerelease version, they did a lexicographical sorting which would put `-nightly10` *before* `-nightly9` in the sort order.

NuGet's codebase was [forked](https://github.com/anaisbetts/NuGet/) to integrate it into Squirrel.Windows so I was able to send a PR there to [add support for double-digit prerelease version sorting](https://github.com/anaisbetts/NuGet/pull/1).  Unfortunately, that wasn't enough as [there's a *second* SemVer 1.0 implementation](https://github.com/Squirrel/Squirrel.Windows/blob/develop/src/StubExecutable/Semver200_comparator.cpp) used in Squirrel.Windows in the app launcher shim, written in C++.  @Ben3eee [started a change](https://github.com/Ben3eeE/Squirrel.Windows/commit/44dfd3729711193b37025f24e4b81ba43c90ec6f) to fix this behavior but neither of us have had time to complete it.

**Comment by rafeca** [comment]
We've fixed this issue by cherry-picking the PR from @Ben3eeE to create a custom build of `electron-winstaller`.

With this, current Atom nightlies don't experience this issue, so I'm going to close it 😃
